internal/controller/authenticator/webhookcachefiller/webhookcachefiller_test.go

```diff
@@ -100,59 +928,159 @@ func TestController(t *testing.T) {
+				}
+			} else {
+				require.Error(t, errors.New("wantActions is required for test "+tt.name))
+			}
+
+			require.Equal(t, tt.wantCacheEntries, len(cache.Keys()), fmt.Sprintf("expected cache entries is incorrect. wanted:%d, got: %d, keys: %v", tt.wantCacheEntries, len(cache.Keys()), cache.Keys()))
 		})
 	}
 }
 
 func TestNewWebhookAuthenticator(t *testing.T) {
-	t.Run("temp file failure", func(t *testing.T) {
+	goodEndpoint := "https://example.com"
+
+	t.Run("prerequisites not ready, cannot create webhook authenticator", func(t *testing.T) {
+		conditions := []*metav1.Condition{}
+		res, conditions, err := newWebhookAuthenticator(&auth1alpha1.WebhookAuthenticatorSpec{}, os.CreateTemp, clientcmd.WriteToFile, conditions, false)
+		require.Equal(t, []*metav1.Condition{
+			{
+				Type:    "AuthenticatorValid",
+				Status:  "Unknown",
+				Reason:  "UnableToValidate",
+				Message: "unable to validate; see other conditions for details",
+			},
+		}, conditions)
+		require.Nil(t, res)
+		require.Nil(t, err)
+	})
+
+	t.Run("temp file failure, cannot create webhook authenticator", func(t *testing.T) {
 		brokenTempFile := func(_ string, _ string) (*os.File, error) { return nil, fmt.Errorf("some temp file error") }
-		res, err := newWebhookAuthenticator(nil, brokenTempFile, clientcmd.WriteToFile)
+		conditions := []*metav1.Condition{}
+		res, conditions, err := newWebhookAuthenticator(nil, brokenTempFile, clientcmd.WriteToFile, conditions, true)
+		require.Equal(t, []*metav1.Condition{
+			{
+				Type:    "AuthenticatorValid",
+				Status:  "False",
+				Reason:  "UnableToCreateTempFile",
+				Message: "unable to create temporary file: some temp file error",
+			},
+		}, conditions)
 		require.Nil(t, res)
 		require.EqualError(t, err, "unable to create temporary file: some temp file error")
 	})
 
-	t.Run("marshal failure", func(t *testing.T) {
+	t.Run("marshal failure, cannot create webhook authenticator", func(t *testing.T) {
 		marshalError := func(_ clientcmdapi.Config, _ string) error { return fmt.Errorf("some marshal error") }
-		res, err := newWebhookAuthenticator(&auth1alpha1.WebhookAuthenticatorSpec{}, os.CreateTemp, marshalError)
+		conditions := []*metav1.Condition{}
+		res, conditions, err := newWebhookAuthenticator(&auth1alpha1.WebhookAuthenticatorSpec{}, os.CreateTemp, marshalError, conditions, true)
+		require.Equal(t, []*metav1.Condition{
+			{
+				Type:    "AuthenticatorValid",
+				Status:  "False",
+				Reason:  "UnableToMarshallKubeconfig",
+				Message: "unable to marshal kubeconfig: some marshal error",
+			},
+		}, conditions)
 		require.Nil(t, res)
 		require.EqualError(t, err, "unable to marshal kubeconfig: some marshal error")
 	})
 
-	t.Run("invalid base64", func(t *testing.T) {
-		res, err := newWebhookAuthenticator(&auth1alpha1.WebhookAuthenticatorSpec{
-			Endpoint: "https://example.com",
+	// t.Run("load kubeconfig err, not currently tested, may not be necessary to test?")
```

Comment:
This looks like a TODO?

Reply:
It's a note that we may want to test, I'll change it to:
```
// t.Run("load kubeconfig err, not currently tested")
```
The production code is right in the middle of that big refactor comment:

```
// TODO refactor this code to directly construct the rest.Config
//  ideally we would keep rest config generation contained to the kubeclient package
//  but this will require some form of a new WithTLSConfigFunc kubeclient.Option
//  ex:
//  _, caBundle, err := pinnipedauthenticator.CABundle(spec.TLS)
//  ...
```
I'm not touching that chunk of code as part of this PR.